src/kirby/components/modal/services/modal.controller.ts

```diff
@@ -1,33 +1,37 @@
 import { Injectable, ViewContainerRef } from '@angular/core';
 
-import { ModalHelper } from './modal.helper';
-import { ModalConfig } from '../config/modal-config';
+import { ModalWindowHelper } from './modal-window.helper';
+import { ActionSheetHelper } from './action-sheet.helper';
+import { ModalWrapperConfig } from '../modal-wrapper/config/modal-wrapper-config';
 import { IModalController } from './modal.controller.interface';
+import { ActionSheetConfig } from '../action-sheet/config/action-sheet-config';
 
 @Injectable()
 export class ModalController implements IModalController {
   private modals: { close: (data?: any) => {} }[] = [];
+  // registerModal needs to be wrapped, because it is a function with side-effects (we modify this.modals),
+  // hence this.modals.push(modal) is going to throw an error once we invoke it from another class
+  private registerModalWrapper: (modal: { close: (data?: any) => {} }) => void = (modal) => {
+    this.register(modal);
+  };
 
-  constructor(private modalHelper: ModalHelper) {}
+  constructor(
+    private modalWindowHelper: ModalWindowHelper,
+    private actionSheetHelper: ActionSheetHelper
+  ) {}
 
-  public showModal(
-    config: ModalConfig,
+  public openModal(
+    config: ModalWrapperConfig,
     vcRef: ViewContainerRef,
     onCloseModal?: (data?: any) => void
   ): void {
-    // registerModal needs to be wrapped, because it is a function with side-effects (we modify this.modals),
-    // hence this.modals.push(modal) is going to throw an error once we invoke it from another class
-    const registerModalWrapper: (modal: { close: (data?: any) => {} }) => void = (modal) => {
-      this.registerModal(modal);
-    };
-
-    const modalCloseEvent: Promise<any> = this.modalHelper.showModal(
+    const modalCloseEvent: Promise<any> = this.modalWindowHelper.showModalWindow(
       config,
       vcRef,
-      registerModalWrapper
+      this.registerModalWrapper
```

Comment:
Just bind the method to `this`:
```suggestion
      this.register.bind(this)
```

Reply:
this is cool ! fixed